### PR TITLE
ZOSVPC-3065 Update the readme file for Standard and Quickstart Variation.

### PR DESCRIPTION
**Pull request by Nikhil22February**
Update the readme file for Deployable Architecture main repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,78 +11,31 @@
...
+    - Virtual Server Instances for every subnet.
+    - Increases security with Key Management.
+    - Collects and stores Internet Protocol (IP) traffic information with Activity Tracker and Flow Logs.
+    - Securely connects to multiple networks with a client-to-site/site-to-site virtual private network
+    - Simplifies risk management and demonstrates regulatory compliance with Financial Services
+    - Uses an edge VPC for secure access through the public internet.
 <!-- Remove the content in this previous H2 heading -->
 ## Reference architectures
+- [Quickstart Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/standard_quickstart/solutions/quickstart)
+- [Standard Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/standard_quickstart/solutions/standard)
```
Comment (kevinkuhner) [inline README.md:38]: The links for Quickstart and Standards variations are pointing to the standard_quickstart branch but will eventually need to be updated to point to the main branch.

File: README.md [hunk 2]
```diff
@@ -91,78 +44,42 @@ See "Reference architecture" in Authoring Guidelines in the public documentation
...
-Add an example of the use of the module in the following code block.
-
-Use real values instead of "var.<var_name>" or other placeholder values
-unless real values don't help users know what to change.
--->
-
-```hcl
-
-```
+<!-- BEGIN OVERVIEW HOOK -->
```
Comment (kevinkuhner) [inline README.md:47]: What's the difference between the summary and overview sections?  

In the overview - why have a link back to the same module (1st link below).  

I suggest reviewing the README format from the module template and adhering to that as well as looking at other mature modules as examples.

File: README.md [hunk 3]
```diff
@@ -91,78 +44,42 @@ See "Reference architecture" in Authoring Guidelines in the public documentation
...
+## Overview
+* [terraform-ibm-zvsi](#terraform-ibm-zvsi)
+* [Submodules](./)
```
Comment (kevinkuhner) [inline README.md:50]: What is the purpose of the Submodules link?

File: README.md [hunk 4]
```diff
@@ -91,78 +44,42 @@ See "Reference architecture" in Authoring Guidelines in the public documentation
...
 ### Requirements
 
 | Name | Version |
 |------|---------|
-| <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.0.0, <1.6.0 |
-
-### Modules
-
-No modules.
+| <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.3, < 1.6 |
```
Comment (kevinkuhner) [inline README.md:78]: Wondering what the value of linking back to this same section is here??

File: README.md [hunk 5]
```diff
@@ -91,78 +44,42 @@ See "Reference architecture" in Authoring Guidelines in the public documentation
...
-        - **Sample Cloud Service** service
-            - `Administrator` platform access
--->
-
-<!-- NO PERMISSIONS FOR MODULE
-If no permissions are required for the module, uncomment the following
-statement instead the previous block.
--->
+            - `Editor` platform access
+        - **VPC Infrastructure Services** service
```
Comment (kevinkuhner) [inline README.md:65]: Check with Akshay on the complete list of permissions required for this module and update here.

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Nikhil22February - please see my comments and let me know if you have any questions.  Please work with @Pketaki  and Akshay to get the required info that should be in the README.

File: README.md [hunk 6]
```diff
@@ -11,78 +11,31 @@
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md:11]: Updated with the latest release badge with this module.

File: README.md [hunk 7]
```diff
@@ -11,78 +11,31 @@
...
-
-### Post-merge steps
-
-After the first PR for your module is merged, follow these post-merge steps:
-
-1.  Create a PR to enable the upgrade test by removing the `t.Skip` line in `tests/pr_test.go`.
-
+## Summary
+
+This repository contains wazi deployable architecture solutions that help provision VPC landing zones and interconnect them. The below solutions are available in the IBM Cloud Catalog and can also be deployed with terraform.
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested.

File: README.md [hunk 8]
```diff
@@ -11,78 +11,31 @@
...
+
+Two solutions are offered:
+1. [Quickstart Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/standard_quickstart/solutions/quickstart)
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested after it will get merge into the main branch the link can be accessible.

File: README.md [hunk 9]
```diff
@@ -11,78 +11,31 @@
...
+    This pattern deploys the following infrastructure:
+    - Increases security with Key Management.
+    - Collects and stores Internet Protocol (IP) traffic information with Activity Tracker and Flow Logs.
+    - Simplifies risk management and demonstrates regulatory compliance with Financial Services
+    - Uses an edge VPC for secure access through the public internet.
+2. [Standard Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/standard_quickstart/solutions/standard)
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested after it will get merge into the main branch the link can be accessible.

File: README.md [hunk 10]
```diff
@@ -11,78 +11,31 @@
...
-
-1.  Create a PR to enable the upgrade test by removing the `t.Skip` line in `tests/pr_test.go`.
-
+## Summary
+
+This repository contains wazi deployable architecture solutions that help provision VPC landing zones and interconnect them. The below solutions are available in the IBM Cloud Catalog and can also be deployed with terraform.
+
+Two solutions are offered:
+1. [Quickstart Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/standard_quickstart/solutions/quickstart)
+    This pattern deploys the following infrastructure:
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md:20]: Made changes as suggested.

File: README.md (hunk 1 above)
Comment (Nikhil22February) [reply README.md]: Made changes as suggested after it will get merge into the main branch the link can be accessible.

File: README.md (hunk 2 above)
Comment (Nikhil22February) [reply README.md]: Removed this section.

File: README.md (hunk 4 above)
Comment (Nikhil22February) [reply README.md]: Removed hyperlink.

File: README.md (hunk 5 above)
Comment (Nikhil22February) [reply README.md]: Made changes as suggested by Akshay to take the reference from IBM cloud permissions for quickstart and standard variation.

File: README.md [hunk 11]
```diff
@@ -5,164 +5,134 @@
...
+| [vpn_client_access_group_users] | List of users in the Client to Site VPN Access Group | `list(string)` | [] | no      standard |
+| [access_group_name] | Name of the IAM Access Group to create if var.create_policy is true | `string` | client-to-site-vpn-access-group | no | standard |
+| [root_ca_max_ttl] | Maximum TTL value for the root CA | `string` | "8760h" | yes | standard |
+| [root_ca_common_name] | Fully qualified domain name or host domain name for the certificate to be created | `string` | "cloud.ibm.com" | no | standard |
+| [resource_group] | Name of the resource group to use for this example. If not set, a resource group is created | `string` | null | no | standard |
+| [cert_common_name] | Fully qualified domain name or host domain name for the private certificate to be created | `string` | n/a | no | standard |
+| [override_json_string] | Override default values with custom JSON. Any value other than an empty string will override all other configuration changes | `string` | "" | no | quickstart, standard |
+| [override] | Override default values with custom JSON template | `bool` | true | no | standard |
```
Comment (kevinkuhner) [inline README.md:124]: Is there any relationship between override_json_string and override?   If override is true, where is the custom JSON template that will be used?  @Pketaki ^^

Add a sentence to explain the relationship and/or difference between these last two variables.

File: README.md [hunk 12]
```diff
@@ -5,164 +5,134 @@
...
+| [time_sleep](https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep) | resource | standard |
+| [ibm_is_subnet](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_subnet_reserved_ip) | data source | standard |
+| [ibm_is_vpc](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_vpc_routing_table_route) | data source | standard |
+| [ibm_is_security_group](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_security_group_rule) | data source | quickstart, standard |
+| [ibm_is_security_group_rule](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_security_group_rule) | resource | quickstart, standard |
+
 
 ### Inputs
 
-No inputs.
```
Comment (kevinkuhner) [inline README.md:160]: Any reason for the order of the rows in the input table?  Should they be in alphabetical order to make it easier to read?    What does the module template readme instructions say?   How do other published modules organize their input variable lists?

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Nikhil22February @Pketaki  - I still have some comments.   Please see latest comments.  Thanks.

File: README.md [hunk 13]
```diff
@@ -5,164 +5,134 @@
 <!-- UPDATE BADGE: Update the link for the following badge-->
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md:5]: Updated title with "Wazi as a service VSI on VPC Landing Zone"

File: README.md [hunk 14]
```diff
@@ -5,164 +5,134 @@
...
+Two solutions are offered:
+1. [Quickstart Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/main/solutions/quickstart)
+    This pattern deploys the following infrastructure:
+    - Workload VPC with Wazi as a service VSI.
+    - Uses Floating IP addresses for access through the public internet.
+2. [Standard Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/main/solutions/standard)
+    This pattern deploys the following infrastructure:
+    - Separate VPC for edge.
+    - Separate VPC for workloads.
+    - Virtual Server Instances for every subnet.
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Modified as suggested.

File: README.md [hunk 15]
```diff
@@ -5,164 +5,134 @@
...
 
-Use real values instead of "var.<var_name>" or other placeholder values
-unless real values don't help users know what to change.
--->
-
-```hcl
-
-```
+## Reference architectures
+- [Quickstart Variation](https://github.com/terraform-ibm-modules/terraform-ibm-zvsi/tree/main/solutions/quickstart)
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md:50]: Yes both are pointing to the same place since I followed the template [module link](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/blob/main/README.md)

File: README.md [hunk 16]
```diff
@@ -5,164 +5,134 @@
...
+| secrets_manager_private_certificate | [terraform-ibm-modules/terraform-ibm-secrets-manager-private-cert](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager-private-cert) | 1.0.2 |
+| client_to_site_vpn | [terraform-ibm-modules/terraform-ibm-client-to-site-vpn](https://github.com/terraform-ibm-modules/terraform-ibm-client-to-site-vpn) | 1.6.2 |
 
 ### Resources
 
-No resources.
+| Name | Type | Variation |
+|------|------|-----------|
+| [time_sleep](https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep) | resource | standard |
+| [ibm_is_subnet](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_subnet_reserved_ip) | data source | standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested.

File: README.md [hunk 17]
```diff
@@ -5,164 +5,134 @@
...
+| [ibm_is_vpc](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_vpc_routing_table_route) | data source | standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested

File: README.md [hunk 18]
```diff
@@ -5,164 +5,134 @@
...
+| [ibm_is_security_group](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_security_group_rule) | data source | quickstart, standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Removed the line #97

File: README.md [hunk 19]
```diff
@@ -5,164 +5,134 @@
...
+| [ibm_is_security_group_rule](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/is_security_group_rule) | resource | quickstart, standard |
+
 
 ### Inputs
 
-No inputs.
+| Name | Description | Type | Default | Required | Variation |
+|------|-------------|------|---------|:--------:|-----------|
+| [ibmcloud_api_key] | The IBM Cloud platform API key needed to deploy IAM enabled resources | `string` | unique | yes | quickstart, standard |
+| [machine_type] | Valid machine type such as "mz2o-2x16", "bz2-4x16", "bz2-8x32", etc... | `string` | "mz2o-2x16" | yes | quickstart, standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: updated with the description and also updated the default value.

File: README.md [hunk 20]
```diff
@@ -5,164 +5,134 @@
...
+| [prefix] | A unique identifier for resources | `string` | n/a | yes | quickstart, standard |
+| [region] | Region where VPC will be created | `string` | n/a | yes | quickstart, standard |
+| [ssh_public_key] | A public SSH Key for VSI creation which does not already exist in the deployment region | `string` | unique | yes | quickstart, standard |
+| [ports] | Inbound port for telnet & zosmf web browser for Wazi VSI SG  | `list(number)` | n/a | yes | quickstart, standard |
+| [image_name] | valid image name for Wazi VSI | `string` | "ibm-zos-2-5-s390x-dev-test-wazi-7" | yes | quickstart, standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested.

File: README.md [hunk 21]
```diff
@@ -5,164 +5,134 @@
...
+| [sm_service_plan] | Type of service plan to use to provision Secrets Manager if not using an existing one | `string` | "standard" | no | standard |
+| [root_ca_name] | Name of the Root CA to create for a private_cert secret engine. Only used when var.existing_sm_instance_guid is false | `string` | "root-ca" | no | standard |
+| [intermediate_ca_name] | Name of the Intermediate CA to create for a private_cert secret engine. Only used when var.existing_sm_instance_guid is false | `string` | "intermediate-ca" | no | standard |
+| [certificate_template_name] | Name of the Certificate Template to create for a private_cert secret engine. When var.existing_sm_instance_guid is true, then it has to be the existing template name that exists in the private cert engine | `string` | "my-template" | no | standard |
+| [create_policy] | Set to true to create a new access group (using the value of var.access_group_name) with a VPN Client role | `bool` | true | no | standard |
+| [vpn_client_access_group_users] | List of users in the Client to Site VPN Access Group | `list(string)` | [] | no      standard |
```
[the remark this answers is not in the document]
Comment (Nikhil22February) [reply README.md]: Made changes as suggested.

File: README.md (hunk 12 above)
Comment (Nikhil22February) [reply README.md:160]: Followed the [template module link](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/blob/main/README.md)

File: README.md (hunk 11 above)
Comment (kevinkuhner) [reply README.md:123]: @Nikhil22February - I think you missed this one.

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Nikhil22February  - there is just one comment that you did not address from last time.  Please see my previous comment regarding override and override_json variables.

File: README.md (hunk 11 above)
Comment (Nikhil22February) [reply README.md:123]: When **override** value is true so we are updating the string and adding it to **override_json** file.

Comment (kevinkuhner) [reply README.md:123]: @Nikhil22February  - I don't understand your answer.   If I set override_json_string to a non-null value, I know that means that will be used to override any other values.  However, what then does the override boolean mean??    How does it relate to override_json_string if at all?  If I'm using this and I set a value for override_json_string do I also need to set override to true?  The descriptions for these two parameters need to be clarified so that a user would know how to use them.

**Review by kevinkuhner — CHANGES_REQUESTED** [review]
@Nikhil22February  - see my latest comment.

File: README.md (hunk 11 above)
Comment (kevinkuhner) [reply README.md:123]: @Nikhil22February  - ok, then I think we need to add to the description of override variable to indicate that if it is true, then the content of override.json file will be used as input to the terraform.  The description of the variable in the variables.tf file is more complete than the README.  It says: "Override default values with custom JSON template. This uses the file `override.json` to allow users to create a fully customized environment."  

Also, should the default value really be true for this variable?   @Pketaki

Comment (Pketaki) [reply README.md:123]: @kevinkuhner Yes. Default should be true as we will use json file as input first and then modified data will be passed as input to override_json_string

Comment (kevinkuhner) [reply README.md:123]: @Nikhil22February @Pketaki  - I still think we need an update to the description for the `override` variable.  Make it match the description that is in the variables.tf file.

Comment (Nikhil22February) [reply README.md]: updated description for **override** and arranged alphabetically.

**Review by kevinkuhner — APPROVED** [review]
@Nikhil22February - looks fine to me.
